### PR TITLE
Moves requirements inside CE so it can be use in standard

### PR DESCRIPTION
**Pull request by BitOne**
**Description (for Contributor and Core Developer)**

This PR aims to put the PIM technical requirements definition inside the src/ directory so it can be re-used by CE standard and EE dev and standard.

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | -
| Added legacy Behats               | -
| Added acceptance tests            | -
| Added integration tests           | -
| Changelog updated                 | Todo
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | -
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Akeneo/Platform/Requirement.php [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+namespace Akeneo\Platform;
+
+/**
+ * Simple value object to transfer requirements information to the requirement checker system
+ *
+ * @author    Benoit Jacquemont <benoit@akeneo.com>
+ * @copyright 2018 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class Requirement
```
Comment (ahocquard) [inline src/Akeneo/Platform/Requirement.php]: why not using `RequirementCollection` ad `Requirement` from Doctrine?

Comment (ahocquard) [reply src/Akeneo/Platform/Requirement.php]: Seen IRL: class files copied from vendor and can't load it with autoload (not following PSR, no namepsace).
So, easier to handle it this way.

File: src/Akeneo/Platform/CommunityRequirements.php [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+    const LOWEST_REQUIRED_MYSQL_VERSION = '5.7.22';
+    const GREATEST_REQUIRED_MYSQL_VERSION = '5.8.0';
+
+    const REQUIRED_EXTENSIONS = [
+        'apcu',
+        'bcmath',
+        'curl',
+        'gd',
+        'intl',
+        'mcrypt',
```
Comment (damien-carcel) [inline src/Akeneo/Platform/CommunityRequirements.php]: to remove.

Comment (BitOne) [reply src/Akeneo/Platform/CommunityRequirements.php]: Thanks!

**Pull request merged**